src/linter/block.go

```diff
@@ -1782,45 +1782,47 @@ func (b *BlockWalker) handleAssign(a *ir.Assign) bool {
 }
 
 func (b *BlockWalker) handleAssignOp(o ir.Node) (res bool) {
-	// We are converting "$x += $y" to "$x = $x + $y"
-	// to reuse existing processing code.
 	switch s := o.(type) {
 	case *ir.AssignPlus:
-		as := ir.Assign{
-			Variable: s.Variable,
-			Expression: &ir.PlusExpr{
-				Left:  s.Variable,
-				Right: s.Expression,
-			},
-		}
-		as.Walk(b)
+		s.Variable.Walk(b)
+		s.Expression.Walk(b)
+		as := &ir.PlusExpr{
+			Position: s.Position,
+			Left:     s.Variable,
+			Right:    s.Expression,
+		}
+		tp := solver.ExprTypeLocal(b.ctx.sc, b.r.ctx.st, as)
+		b.replaceVar(s.Variable, tp, "", meta.VarAlwaysDefined)
 	case *ir.AssignMinus:
-		as := ir.Assign{
-			Variable: s.Variable,
-			Expression: &ir.MinusExpr{
-				Left:  s.Variable,
-				Right: s.Expression,
-			},
-		}
-		as.Walk(b)
+		s.Variable.Walk(b)
+		s.Expression.Walk(b)
+		as := &ir.MinusExpr{
+			Position: s.Position,
+			Left:     s.Variable,
+			Right:    s.Expression,
+		}
+		tp := solver.ExprTypeLocal(b.ctx.sc, b.r.ctx.st, as)
+		b.replaceVar(s.Variable, tp, "", meta.VarAlwaysDefined)
```

Comment:
Don't leave the reason empty.
You can probably move `b.replaceVar` below the switch statement if all cases end with the same code path.
`tp` variable will go above the switch.